pgstmt/select.go

```diff
@@ -167,6 +178,16 @@ func (st *selectStmt) Offset(n int64) {
 func (st *selectStmt) make() *buffer {
 	var b buffer
 	b.push("select")
+	if st.distinct != nil {
+		b.push("distinct")
+
+		if st.distinct.on {
```

Comment:
check col empty instead of `on` here because
`select distinct on () *` is invalid.